benchmarks/src/path_lexically_normal.cpp

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <benchmark/benchmark.h>
+#include <filesystem>
+#include <string_view>
+
+namespace {
+    void BM_lexically_normal(benchmark::State& state) {
+        using namespace std::literals;
+        static constexpr std::wstring_view args[5]{
+            LR"(X:DriveRelative)"sv,
+            LR"(\\server\\\share)"sv,
+            LR"(STL/.github/workflows/../..)"sv,
+            LR"(C:\Program Files\Azure Data Studio\resources\app\extensions\bat\snippets\batchfile.code-snippets)"sv,

```

Comment:
Is it better to replace this test case with a more generic one?

Reply:
Seems fine to me since it's a realistic Microsoft path. If it were from some other program then I'd request a generic meow-style example.

Reply:
yes, it is a real path in my computer.
for `C:\Program Files\Azure Data Studio...` I find a better one: 
`C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\VC\Snippets` (it has the same number of segments, and is shorter than the last test case(`/\server/\share/\a...`).

Reply:
If you want to add more tests, it's probably better to do it in a new PR to not reset the review process.

Reply:
Generally, when a PR has been moved to Ready To Merge, changes should only be pushed when they're big enough to be worth it (fixing bugs or major omissions), or they're small enough to be unquestionably safe like comment typo fixes. You should ping the maintainers who approved, on GitHub and/or Discord, so they notice and can take another look.

Basically, we maintainers love doing work, but we want to focus it on necessary work that keeps PRs flowing and the codebase moving forward. We try to minimize avoidable work, so we prefer to avoid scenarios like:

* When a PR that was ready to go is changed in a way that requires further rework.
  + If the change was an attempt to fix a real bug, then this is not annoying - we'd rather pull a PR at the last minute than merge a bug. However, if the change wasn't really necessary, then iterating on it again is extra work.
* When changes are pushed after Ready To Merge without maintainers being notified.
  + This is a potential process loophole so we don't like it - we require all codebase modifications to be reviewed by another maintainer so last-minute changes that try to sneak in are not cool. (This is applied uniformly; if maintainer X creates a PR, and maintainer Y approves, then X doesn't get to push further changes without Y taking a look. The only exceptions are fixing build breaks and test failures during mirroring, in which case we notify but don't block on getting re-approval.)
* When changes are pushed after a maintainer has self-assigned the PR and commented "I'm mirroring this to the MSVC-internal codebase, please notify me of any changes", and they aren't for a *really* good reason.
  + Resetting this mirroring process is extra extra work and adds a lot of delay.

Reply:
hm, thanks for clarification 👀